app/forms/provider_interface/offer_wizard.rb

```diff
@@ -90,21 +92,27 @@ def save_and_go_to_next_step(step)
       next_step(step)
     end
 
+    def query_service
+      @query_service ||= GetChangeOfferOptions.new(
+        user: ProviderUser.find(provider_user_id),
+        current_course: ApplicationChoice.find(application_choice_id).offered_course,
```

Comment:
Can we extract provider_user and offered_course to private reusable methods.

Also can this method definition be moved further down so the private methods accessed by the public offer wizard methods are more prominent.